From cba514d5ab670aea99217a3d4447a7877adca646 Mon Sep 17 00:00:00 2001
From: Anuradha Bathina <quic_abathina@quicinc.com>
Date: Wed, 3 Nov 2021 21:13:56 +0530
Subject: [PATCH] Send_data: Fix for Timeout Error in DIAG output.

Instead of waiting for timeout, print response data corresponding to sent msg only.
---
 tools/send_data.c | 43 ++++++++++++++++++++++++++++++++++++++++---
 1 file changed, 40 insertions(+), 3 deletions(-)

diff --git a/tools/send_data.c b/tools/send_data.c
index 84a338b..1ea338e 100644
--- a/tools/send_data.c
+++ b/tools/send_data.c
@@ -28,6 +28,40 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  */
+/*
+Changes from Qualcomm Innovation Center are provided under the following license:
+Copyright (c) 2021 Qualcomm Innovation Center, Inc. All rights reserved.
+ 
+Redistribution and use in source and binary forms, with or without
+modification, are permitted (subject to the limitations in the
+disclaimer below) provided that the following conditions are met:
+ 
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+ 
+    * Redistributions in binary form must reproduce the above
+      copyright notice, this list of conditions and the following
+      disclaimer in the documentation and/or other materials provided
+      with the distribution.
+ 
+    * Neither the name of Qualcomm Innovation Center, Inc. nor the names of its
+      contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+ 
+NO EXPRESS OR IMPLIED LICENSES TO ANY PARTY'S PATENT RIGHTS ARE
+GRANTED BY THIS LICENSE. THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT
+HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED
+WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE
+GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN
+IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
 
 #include <sys/socket.h>
 #include <sys/un.h>
@@ -55,7 +89,7 @@ typedef uint32_t        uint32;
 int main(int argc, char **argv)
 {
 	struct sockaddr_un addr;
-	unsigned char *msg;
+	unsigned char msg;
 	struct timeval tv = {5, 0};
 	fd_set rfds;
 	ssize_t n;
@@ -87,6 +121,7 @@ int main(int argc, char **argv)
 	for (i = 4; i < REQ_LOOPBACK_LEN; i++,j++)
 		req_modem_loopback[i] = atoi(argv[j]);
 
+	msg = req_modem_loopback[4];
 
 	fd = socket(AF_UNIX, SOCK_SEQPACKET, 0);
 	if (fd < 0)
@@ -155,10 +190,12 @@ int main(int argc, char **argv)
 		printf("%02x ", ptr[i]);
 	}
 
-		if (buf[0] == msg[0])
+		if (ptr[0] == msg){
+			printf("\nSend_data: Reading Response for the sent msg is completed.\n");
 			break;
+		}
 	}
-
+	close(fd);
 	return 0;
 }
 
-- 
2.17.1

